From cda51ac821166c50e30f5dabedd54fdece2535c3 Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Wed, 18 Jan 2017 17:36:29 -0500
Subject: [PATCH 5305/5855] drm/amdkfd: Preempt static queues on eviction

Eviction needs to preempt all queues, including static queues.
Processes in the GPU debugger will cause VM faults on evictions.

This has potential side effects for the GPU debugger, and may not
work at all for queues that have halted waves. A better solution
is needed to ensure seamless preemption with the GPU debugger.

Change-Id: I706c85666258d180adc06e0487d95fb1425a82bd
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
index 2b5eaa9..f0c97b3 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
@@ -528,7 +528,7 @@ int process_evict_queues(struct device_queue_manager *dqm,
 			dqm->queue_count--;
 	}
 	if (dqm->sched_policy != KFD_SCHED_POLICY_NO_HWS)
-		retval = execute_queues_cpsch(dqm, false);
+		retval = execute_queues_cpsch(dqm, qpd->is_debug);
 
 	mutex_unlock(&dqm->lock);
 	return retval;
-- 
2.7.4

